core/common/src/test/scala/org/typelevel/otel4s/TextMapGetterProps.scala

```diff
@@ -0,0 +1,99 @@
+import org.scalacheck.Arbitrary
+import org.scalacheck.Prop.forAll
+
+import scala.collection.concurrent
+import scala.collection.generic.IsIterable
+import scala.collection.immutable
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+class TextMapGetterProps extends ScalaCheckSuite {
+  def mkProps[A: Arbitrary](implicit
+      tag: ClassTag[A],
+      getter: TextMapGetter[A],
+      conv: IsIterable[A] { type A = (String, String) }
+  ): Unit = {
+    val className =
+      tag.runtimeClass match {
+        case cls if cls.isArray => "Array"
+        case cls                => cls.getName
+      }
+    property(s"TextMapSetter#get  for $className") {
+      forAll { (coll: A, key: String) =>
+        assertEquals(
+          getter.get(coll, key),
+          conv(coll).find(_._1 == key).map(_._2)
+        )
+      }
+    }
+    property(s"TextMapSetter#keys for $className") {
+      forAll { (coll: A) =>
+        assertEquals(getter.keys(coll).toSet, conv(coll).view.map(_._1).toSet)
+      }
+    }
+  }
+
+  mkProps[collection.Map[String, String]]
+  mkProps[collection.SortedMap[String, String]]
+  mkProps[collection.SeqMap[String, String]]
+  mkProps[immutable.Map[String, String]]
+  mkProps[immutable.HashMap[String, String]]
+  mkProps[immutable.SortedMap[String, String]]
+  mkProps[immutable.TreeMap[String, String]]
+  mkProps[immutable.ListMap[String, String]]
+  mkProps[immutable.VectorMap[String, String]]
+  mkProps[immutable.TreeSeqMap[String, String]]
+  mkProps[mutable.Map[String, String]]
+  mkProps[mutable.HashMap[String, String]]
+  mkProps[mutable.CollisionProofHashMap[String, String]]
+  mkProps[mutable.SortedMap[String, String]]
+  mkProps[mutable.TreeMap[String, String]]
+  mkProps[mutable.SeqMap[String, String]]
+  mkProps[mutable.LinkedHashMap[String, String]]
+  mkProps[concurrent.TrieMap[String, String]]
+
+  mkProps[collection.Seq[(String, String)]]
+  mkProps[collection.LinearSeq[(String, String)]]
+  mkProps[collection.IndexedSeq[(String, String)]]
+  mkProps[immutable.Seq[(String, String)]]
+  mkProps[immutable.LinearSeq[(String, String)]]
+  mkProps[immutable.List[(String, String)]]
+  mkProps[immutable.LazyList[(String, String)]]
+  mkProps[immutable.Queue[(String, String)]]
+  mkProps[immutable.IndexedSeq[(String, String)]]
+  mkProps[immutable.Vector[(String, String)]]
+  mkProps[immutable.ArraySeq[(String, String)]]
+  mkProps[mutable.Seq[(String, String)]]
+  mkProps[mutable.IndexedSeq[(String, String)]]
+  mkProps[mutable.ArraySeq[(String, String)]]
+  mkProps[mutable.ArraySeq[(String, String)]]
+  mkProps[mutable.Buffer[(String, String)]]
+  mkProps[mutable.ListBuffer[(String, String)]]
+  mkProps[mutable.IndexedBuffer[(String, String)]]
+  mkProps[mutable.ArrayBuffer[(String, String)]]
+  mkProps[mutable.Queue[(String, String)]]
+  mkProps[mutable.Stack[(String, String)]]
+  mkProps[mutable.ArrayDeque[(String, String)]]
+  mkProps[mutable.UnrolledBuffer[(String, String)]]
```

Comment:
This is neat.  Very few of these will be used in practice, but just the compilation of it is a nice test that there are no surprises in the resolution of the implicits.